engine/sdk_tx_getter.go

```diff
@@ -0,0 +1,59 @@
+package engine
+
+import (
+	"context"
+	"slices"
+	"time"
+
+	sdk "github.com/bitcoin-sv/go-sdk/transaction"
+	"github.com/bitcoin-sv/spv-wallet/engine/spverrors"
+	"iter"
+)
+
+// loadFromDBTimeout - within this time transactions should be loaded from the database
+const loadFromDBTimeout = 20 * time.Second
+
+type sdkTxGetter struct {
+	client *Client
+}
+
+func newSDKTxGetter(client *Client) *sdkTxGetter {
+	return &sdkTxGetter{client: client}
+}
+
+func (g *sdkTxGetter) GetTransactions(ctx context.Context, ids iter.Seq[string]) ([]*sdk.Transaction, error) {
```

Comment:
🤔 I wonder if we could move it to separate package!?

Reply:
Perhaps yes. But in some next step.